### PR TITLE
[Mellanox] Add SAI template config support

### PR DESCRIPTION
**Pull request by nazariig**
Signed-off-by: Nazarii Hnydyn <nazariig@nvidia.com>

Allow SAI xml dynamic config override when template is defined

### CODE REVIEW

**Review by kcudnik — APPROVED** [review]

**Comment by nazariig** [comment]
@anish-n please have a look

**Comment by liat-grozovik** [comment]
@kcudnik can you please add a label for 202012? We need this one to be part of it

File: syncd/scripts/syncd_init_common.sh [hunk 1]
```diff
@@ -111,6 +111,12 @@ config_syncd_mlnx()
     cat $HWSKU_DIR/sai.profile > /tmp/sai.profile
     echo "DEVICE_MAC_ADDRESS=$MAC_ADDRESS" >> /tmp/sai.profile
     echo "SAI_WARM_BOOT_WRITE_FILE=/var/warmboot/" >> /tmp/sai.profile
+
+    # Use SAI template config if present
+    if [[ -f "${HWSKU_DIR}/sai.xml.j2" ]]; then
+        sonic-cfggen -d -t /usr/share/sonic/hwsku/sai.xml.j2 > /tmp/sai.xml
```
Comment (lguohan) [inline syncd/scripts/syncd_init_common.sh:117]: please combine this with line 16, sonic-cfggen causes latency, we intend to use sonic-cfggen only once per docker.

**Comment by nazariig** [comment]
https://github.com/Azure/sonic-buildimage/pull/7188

File: syncd/scripts/syncd_init_common.sh (hunk 1 above)
Comment (nazariig) [reply syncd/scripts/syncd_init_common.sh]: @lguohan fixed

**Review by anish-n — APPROVED** [review]

**Review by kcudnik — APPROVED** [review]

**Comment by kcudnik** [comment]
if this is good i can merge it

**Comment by nazariig** [comment]
> 
> 
> if this is good i can merge it

@lguohan please have a look

**Comment by nazariig** [comment]
@lguohan just a kind reminder

**Review by lguohan — APPROVED** [review]

**Pull request merged**